lib/client/js/ui/controls/AuthButton.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * The UI Control that manages the AuthButton button component
+ * residing in the Header section
+ *
+ * @class
+ */
+var AuthButton = (function () {
+
+    /**
+     * @alias AuthButton
+     */
+    function AuthButton(config) {
+        // Call the super
+        Control.apply(this, [config]);
+    }
+
+    AuthButton.prototype = new Control();
+
+    AuthButton.prototype.view = function () {
+        if (this.getProp("loggedInUser") === null) {
+            return $("<button>", {"class": "beans--AuthButton"})
+                .text("Login");
+        } else {
+            return $("<button>", {"class": "beans--AuthButton"})
+                .text("Logout");
+        }
+    };
+
+    AuthButton.prototype.attachEvent = function () {
+        var self = this;
+        $(".beans--AuthButton").click(function () {
+            if ($(this).text() === "Login") {
```

Comment:
We should create the decision point in this handler much like the way we handle it in the `view` method. Otherwise, if we change the display of the string in the `view` we'll have to change this method as well.